### ISSUE TITLE
Fix JSON Schema Dependency

### ISSUE THREAD

**Issue by mfeit-internet2** [perfsonar/pscheduler#406]
Adjust the requirement for the Python JSON Schema module to require at least 2.5.0.

Antoine observed this with Debian's 2.3.0:
```
ps-iniu@pS40-z1-d8-12:~$ pscheduler task --debug clock --timeout 30 --dest ps-d8.eshango.net
2017-04-12T11:45:43 Debug signal ignored; already not debugging
2017-04-12T11:45:43 Debug discontinued
2017-04-12T11:45:43 Assistance is from localhost
2017-04-12T11:45:43 Using slip of PT5M
2017-04-12T11:45:43 Converting to spec via https://localhost/pscheduler/tests/clock/spec
Submitting task...
2017-04-12T11:45:47 Fetching participant list
2017-04-12T11:45:47 {"dest": "ps-d8.eshango.net", "timeout": "30", "schema": 1}
Unable to determine the lead participant: Traceback (most recent call last):
  File "/usr/lib/pscheduler/commands/../classes/test/clock/participants", line 15, in <module>
    valid, message = spec_is_valid(json)
  File "/usr/lib/pscheduler/classes/test/clock/validate.py", line 22, in spec_is_valid
    return json_validate(json, schema)
  File "/usr/lib/python2.7/dist-packages/pscheduler/jsonval.py", line 776, in json_validate
    '/' + ('/'.join([str(x) for x in ex.absolute_path])),
AttributeError: 'ValidationError' object has no attribute 'absolute_path'
```

**Comment by tonin** [comment]
If it is of any help, [jessie-backports alread has 2.5.1](https://packages.debian.org/jessie-backports/python-jsonschema).  We can probably re-use the `debian/` files.

**Comment by tonin** [comment]
@vvidic Can you have a look at this on Wheezy?  There are more missing libs.  I've added python-functools32 here and a few others in https://github.com/perfsonar/minor-packages/commits/debian/wheezy (not sure there are all needed) but the package still doesn't compile properly.

**Comment by vvidic** [comment]
On Wed, Apr 12, 2017 at 05:38:10PM -0700, Antoine Delvaux wrote:
> @vvidic Can you have a look at this on Wheezy?  There are more missing
> libs.  I've added python-functools32 here and a few others in
> https://github.com/perfsonar/minor-packages/commits/debian/wheezy (not
> sure there are all needed) but the package still doesn't compile
> properly.

It seems that this works:

  pscheduler task clock --dest localhost

but adding the --timeout option breaks things.  Will check what is
missing...

-- 
Valentin Vidic
Computer Systems Engineer - Expert
Department of Computer Infrastructure and Services
Croatian Academic and Research Network - CARNet
Josipa Marohnica 5, HR-10000 Zagreb, Croatia
tel: +385 1 6661 714, fax. +385 1 6661 635
gsm: +385 91 2480 919
www.CARNet.hr

**Comment by tonin** [comment]
The missing bit is to have python-jsonschema in version >= 2.5. I confirmed it yesterday with the package from jessie-backport on Jessie. But we have to port it back to Wheezy. 

Or where you thinking about patching version 2.3.0 in some other ways?

-- Antoine --

**Comment by vvidic** [comment]
On Thu, Apr 13, 2017 at 12:33:41AM -0700, Antoine Delvaux wrote:
> The missing bit is to have python-jsonschema in version >= 2.5. I
> confirmed it yesterday with the package from jessie-backport on
> Jessie. But we have to port it back to Wheezy. 
> 
> Or where you thinking about patching version 2.3.0 in some other ways?

Nope, let's just push whatever is needed to all repos.

-- 
Valentin Vidic
Computer Systems Engineer - Expert
Department of Computer Infrastructure and Services
Croatian Academic and Research Network - CARNet
Josipa Marohnica 5, HR-10000 Zagreb, Croatia
tel: +385 1 6661 714, fax. +385 1 6661 635
gsm: +385 91 2480 919
www.CARNet.hr

**Comment by tonin** [comment]
This is actually fixed for 1.0 both on Jessie and Wheezy (which was a bit harder).
